R/arg.R

```diff
@@ -47,9 +47,23 @@ arg_match <- function(arg, values = NULL) {
   }
 
   arg <- arg[[1]]
+  arg_match_val(arg, values, arg_nm)
+}
+
+#' @description
+#' `arg_match_val()` is a bare-bones version if performance is at a premium.
+#' It requires a string as `arg` and explicit `values`.
+#' @param arg_nm The label to be used for `arg` in error messages.
+#' @rdname arg_match
+#' @export
+arg_match_val <- function(arg, values, arg_nm = as_label(substitute(arg))) {
```

Comment:
The `as_label(substitute(arg))` will cause lots of weird error messages.

Reply:
Do you have an example?

Reply:
Internal calls and internal symbols (even worse if the internal call is complex). I guess it's going to be rare to call this from internal helpers though, so I agree that this default will work well most of the time. I think I like the idea of a bare bones variant, but then it makes sense to have all arguments mandatory? Probably should be called `arg_match0()`? This way we have a general suffix for naming bare bones variants.

Reply:
I would not make `arg_nm` mandatory. It's bare-bones in the sense that it's fast, not that we need to repeat ourselves in the code that uses it if we don't have to.